### PR TITLE
Move files to proper locations. Add idea iml file. Add gradlew

### PR DESCRIPTION
**Pull request by dankwart-de**

### CODE REVIEW

**Review by fkaercher — COMMENTED** [review]
Why did you add idea iml and graddle wrapper etc?

**Comment by dankwart-de** [comment]
When you want to use gradlew, it is recommended to add the gradlew (and gradle?) files to the repository. That's what I found on their website.

I added the RoddyToolLib project to our RoddyInfra project map. For this, the IML is necessary. That's also much closer to the structure we already have.

I also moved the files from src/main/groovy to src and the test files to a test folder. Idea uses this setup and struggled with the former one.

**Comment by dankwart-de** [comment]
You can do the same things for BatchEuphoria! Be aware, that I also renamed the IOHelper class back to RoddyIOHelper. I want to keep things with as few changes as possible now. I know, the suggestion came from me, but I know think it is much better to create as few trouble as possible :)

File: src/main/groovy/de/dkfz/roddy/StringConstants.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
     public static final String SPLIT_SBRACKET_LEFT = "\\[";
     public static final String SPLIT_SBRACKET_RIGHT = "\\[";
 
-    public static final String STOP = ".";
+    public static final String STOP = "";
```
Comment (vinjana) [inline src/main/groovy/de/dkfz/roddy/StringConstants.java]: What is the purpose of an empty string?

**Comment by dankwart-de** [comment]
Could you please re-review?

**Review by fkaercher — APPROVED** [review]

File: src/main/groovy/de/dkfz/roddy/core/InfoObject.java [hunk 2]
```diff
@@ -67,6 +68,7 @@ public void updateTimeStamp() {
 
     public int getAgeInSeconds() {
         long age = (int)(((new Date().getTime()) - timeStamp.getTime()) / 1000);
+//        System.out.println(age);
```
Comment (vinjana) [inline src/main/groovy/de/dkfz/roddy/core/InfoObject.java:71]: Debug code.

File: src/main/groovy/de/dkfz/roddy/execution/io/ExecutionResult.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    public final int errorNumber;
+    /**
+     * All result lines.
+     */
+    public final List<String> resultLines;
+    /**
+     * First line of the result array.
+     * Null if no entries are in the array.
+     */
+    public final String firstLine;
```
Comment (vinjana) [inline src/main/groovy/de/dkfz/roddy/execution/io/ExecutionResult.java:36]: Why is this necessary? Why is this not a method? Also, testing no output by `resultLines.empty()` is much more speaking than `firstLine == null`.

File: src/main/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethods.groovy [hunk 4]
```diff
@@ -162,16 +163,17 @@ class RoddyIOHelperMethods {
...
         try {
-            //String cmd = Roddy.getLocalCommandSet().getCopyDirectoryCommand(src, tgt);
-            ExecutionHelper.executeSingleCommand(cmd);
+            FileUtils.copyDirectory(src, tgt)
+            tgt.setReadable(src.canRead())
+            tgt.setWritable(src.canWrite())
+            tgt.setExecutable(src.canExecute())
+
         } catch (Exception ex) {
             logger.severe(ex.toString())
```
Comment (vinjana) [inline src/main/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethods.groovy:178]: Is this actually the right place to decide, whether a copy failure should be fatal or not? Is that not rather the responsibility of the calling code? It depends on the directory, whether a failure here is a problem.

File: src/main/groovy/de/dkfz/roddy/core/InfoObject.java (hunk 2 above)
Comment (dankwart-de) [reply src/main/groovy/de/dkfz/roddy/core/InfoObject.java:71]: Done

File: src/main/groovy/de/dkfz/roddy/execution/io/ExecutionResult.java (hunk 3 above)
Comment (dankwart-de) [reply src/main/groovy/de/dkfz/roddy/execution/io/ExecutionResult.java:36]: One of my many examples for optimizations :). If you have an accessor and repeatedly call it, I tend to cache values. Same here... though we don't need the cache.

Comment (vinjana) [reply src/main/groovy/de/dkfz/roddy/execution/io/ExecutionResult.java:36]: In this case, I doubt the optimization will actually yield an improvement -- neither for the emptyness check nor for getting the first line is an expensive operation (O(1)).

Comment (dankwart-de) [reply src/main/groovy/de/dkfz/roddy/execution/io/ExecutionResult.java:36]: I'll just change it.

File: src/main/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethods.groovy (hunk 4 above)
Comment (dankwart-de) [reply src/main/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethods.groovy:178]: Well, I know it is different because of the original non-strict Roddy design: "Run as far as you can". In that case it makes sense, but yes for a more public design we will take out the exception here.

Comment (vinjana) [reply src/main/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethods.groovy:178]: Maybe we can add a switch to turn off (not on!) the exception. Thus the function can be used in both contexts -- strict (with exception) and non-strict.
